zipline/pipeline/factors/statistical.py

```diff
@@ -66,11 +80,10 @@ class RollingPearson(_RollingCorrelation):
     Most users should call Factor.pearsonr rather than directly construct an
     instance of this class.
     """
-    def compute(self, today, assets, out, factor_data, slice_data):
-        slice_data_column = slice_data[:, 0]
-        for i in range(len(out)):
-            # pearsonr returns the R-value and the P-value.
-            out[i] = pearsonr(factor_data[:, i], slice_data_column)[0]
+    def compute(self, today, assets, out, factor_data, other_data):
+        super(RollingPearson, self).compute(
+            today, assets, out, factor_data, other_data, pearsonr,
```

Comment:
I think the right way to do this is to forward up the correlation function in `__new__` rather than every time at compute.  As-is, you can construct an instance of the base class that can't actually be computed correctly.